packages/core/core-flows/src/product/workflows/upsert-variant-prices.ts

```diff
@@ -0,0 +1,114 @@
+import { CreateMoneyAmountDTO, ProductTypes } from "@medusajs/types"
+import { Modules, arrayDifference } from "@medusajs/utils"
+import {
+  WorkflowData,
+  createWorkflow,
+  transform,
+} from "@medusajs/workflows-sdk"
+import { removeRemoteLinkStep, useRemoteQueryStep } from "../../common"
+import { createPriceSetsStep, updatePriceSetsStep } from "../../pricing"
+import { createVariantPricingLinkStep } from "../steps"
+
+type WorkflowInput = {
+  variantPrices: {
+    variant_id: string
+    product_id: string
+    prices?: CreateMoneyAmountDTO[]
```

Comment:
q: shouldn't this be Create or UpdateMoneyAmountDTO?

sugg: use `CreatePricesDTO` instead, we should wipe the money amount types eventually.